tests/update_workflow.go

```diff
@@ -157,8 +158,12 @@ func (s *FunctionalSuite) waitUpdateAdmitted(tv *testvars.TestVars, updateID str
 			s.GreaterOrEqual(pollResp.Stage, enumspb.UPDATE_WORKFLOW_EXECUTION_LIFECYCLE_STAGE_ADMITTED)
 			return true
 		}
-		var notFoundErr *serviceerror.NotFound
-		s.ErrorAs(pollErr, &notFoundErr) // poll beat send in race
+		if pollErr.Error() != fmt.Sprintf("update \"%v\" not found", tv.UpdateID(updateID)) {
+			s.T().Log("received error from Update poll: ", pollErr)
+			return true
+		}
```

Comment:
Added this to allow `workflow execution already completed` (also a NotFound) error from stopping the polling.

Reply:
Can you leave type checking too? First check type then message.